internal/server/worker.go

```diff
@@ -0,0 +1,66 @@
+package server
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/arseniy96/bonus-program/internal/logger"
+	"github.com/arseniy96/bonus-program/internal/services/accrual"
+)
+
+const TTL = 1 * time.Second
+
+func (s *Server) OrdersWorker() {
+	ctx, cancel := context.WithTimeout(context.Background(), 3*time.Second)
+	defer cancel()
+
+	for {
+		select {
+		case orderWithTTL := <-s.OrdersQueue:
+			// дёргаем сервис, который отвечает за обработку заказа
+			// внутри сервиса отправляем запрос в accrual
+			// смотрим на статус:
+			// если статус не конечный, то возвращаем order как есть (???)
+			// если статус конечный, то обновляем ордер и создаём запись в bonus_transactions
+			if orderWithTTL.ttl.Before(time.Now()) {
+				res, err := accrual.CheckOrder(s.Config.AccrualHost, orderWithTTL.order.OrderNumber)
+				logger.Log.Info(fmt.Sprintf("%v", res))
+				if err != nil {
+					logger.Log.Errorf("accrual check order error: %v", err)
+					s.OrdersQueue <- OrderWithTTL{
+						order: orderWithTTL.order,
+						ttl:   time.Now().Add(TTL),
+					}
+					continue
+				}
+
+				if res.Status == accrual.OrderStatusRegistered || res.Status == accrual.OrderStatusProcessing {
```

Comment:
самодокументируемый код
```suggestion
				if hasFinalStatus(res) {
```